1020 branch with FFDC and SBE timeout updates

eddiejames opened the pull request.

Commit:
Add FFDC collection

Add a class to watch for the presence of SBEFIFO FFDC reported up
through the OCC hwmon driver.

Signed-off-by: Eddie James <eajames@linux.ibm.com>
Change-Id: Ia8bf6eddb037aec547b72ecce39c1a977d9374bc

cjcain committed:
Use Idle Power Saver parameters from DBus

Allows users to enable and update the IPS parameters instead of
using hardcoded values.

Change-Id: I9010c4b4d3dbdf130a4a778f71c87279681a9f1a
Signed-off-by: Chris Cain <cjcain@us.ibm.com>

Commit:
Add SBE timeout handling

Add an Error instance to detect SBE timeouts in the driver.
Refactor the PLDM code to handle another requeset type (the HRESET) and
another type of sensor event (success or failure of the HRESET).
Handle the timeout and the PLDM event state change in the Manager by
changing the SBE state and requesting an SBE dump if necessary.

Signed-off-by: Eddie James <eajames@linux.ibm.com>
Change-Id: I30d8f9de1914d9808ddb7b547cc57085a5e5779e

geissonator commented:
PR seems to be against the master branch? Do we need a 1020 branch here or can @rfrandse just point to upstream?

Commit:
Add power supply derating factor to meson options

Add the option to specify the power supply derating factor.

Signed-off-by: Eddie James <eajames@linux.ibm.com>
Change-Id: Ibfbb52f64c5b9133fbbc8cbb267aa9618c079cc6

Commit:
Change yamldir to refer to a directory

To maintain the behavior of the autotools build, the "yamldir"
configuration option should refer to the directory in which to find
the occ_sensor.yaml file. Drop the -f argument to the sensor
generation python script and change the default yamldir argument.

Signed-off-by: Eddie James <eajames@linux.ibm.com>
Change-Id: I7a93a0c403c4ce473ea2d7350f3ee441bc55c4be
